packages/web/src/pages/upload-page/fields/TrackMetadataFields.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react'
+
+import { GENRES, getCanonicalName, FeatureFlags } from '@audius/common'
+import { useField } from 'formik'
+
+import DropdownInput from 'components/data-entry/DropdownInput'
+import Input from 'components/data-entry/Input'
+import TagInput from 'components/data-entry/TagInput'
+import TextArea from 'components/data-entry/TextArea'
+import UploadArtwork from 'components/upload/UploadArtwork'
+import { getFeatureEnabled } from 'services/remote-config/featureFlagHelpers'
+import { resizeImage } from 'utils/imageProcessingUtil'
+import { moodMap as untypedMoodMap } from 'utils/moods'
+
+import styles from './TrackMetadataFields.module.css'
+
+const moodMap: Record<string, JSX.Element> = untypedMoodMap
+const MOODS = Object.keys(moodMap).map((k) => ({
+  text: k,
+  el: moodMap[k]
+}))
+
+const messages = {
+  genre: 'Pick a Genre'
+}
+
+type TrackMetadataFieldsProps = {
+  /** Whether or not the preview is playing. */
+  playing: boolean
+  // type: 'track' | 'album' | 'playlist'
+  type: 'track'
+}
+
+const TrackMetadataFields = (props: TrackMetadataFieldsProps) => {
+  const [imageProcessingError, setImageProcessingError] = useState(false)
+  const [artworkField, , artworkHelpers] = useField('artwork')
+  const [titleField, titleMeta, titleHelpers] = useField('title')
+  const [genreField, genreMeta, genreHelpers] = useField('genre')
+  const [, moodMeta, moodHelpers] = useField('mood')
+  const [, tagsMeta, tagsHelpers] = useField('tags')
+  const [, descriptionMeta, descriptionHelpers] = useField('description')
+
+  const onDropArtwork = async (selectedFiles: File[], source: string) => {
+    try {
+      let file = selectedFiles[0]
+      file = await resizeImage(file)
+      const storageV2SignupEnabled = getFeatureEnabled(
```

Comment:
should we use `useFlag` instead? seems to be more used in components.

i think the diff is that the hook can get updates for a given feature flag https://github.com/AudiusProject/audius-client/blob/main/packages/web/src/components/menu/TrackMenu.tsx#L98-L101